components/webui/imports/ui/SearchView/SearchResultsTable.jsx

```diff
@@ -26,18 +31,20 @@ const SearchResultsLoadSensor = ({
             return;
         }
 
-        const observer = new IntersectionObserver(
-            (entries) => {
+        let observer = null;
```

Comment:
I feel it's more intuitive to write something like:

```js
    useEffect(() => {
        if (false === hasMoreResults) {
            return;
        }

        let observer = new IntersectionObserver((entries) => {
            if (entries[0].isIntersecting) {
                loadInterval.current = setInterval(() => {
                    onLoadMoreResults();
                }, SEARCH_RESULTS_LOAD_SENSOR_POLL_INTERVAL_MS);
            } else {
                if (null !== loadInterval.current) {
                    clearInterval(loadInterval.current);
                    loadInterval.current = null;
                }
            }
        });
        observer.observe(loadingBlockRef.current);

        return () => {
            observer.disconnect();
            if (null !== loadInterval.current) {
                clearInterval(loadInterval.current);
            }
        };
    }, [hasMoreResults]);
```

The idea being while the loader is in view there are more results, we periodically try to load more results and when the loader goes out of view, we disable the periodic task. What do you think?

Reply:
Also, I don't think we need the `hasMoreResults` check in the IntersectionObserver handler, right?

Reply:
> Also, I don't think we need the hasMoreResults check in the IntersectionObserver handler, right?

Right, the loading sensor style `visibility` changes to `hidden` when `hasMoreResults === false` and therefore `entries[0].isIntersecting` does not evaluate to `true` in all conditions.

Reply:
> The idea being while the loader is in view and there are more results, we periodically try to load more results and when the loader goes out of view, we disable the periodic task. What do you think?

I agree. The proposed approach is cleaner and avoid unnecessary polling.